### ISSUE TITLE
per-user podcasts

### ISSUE THREAD

**Issue by chrstn-hntschl** [airsonic-advanced/airsonic-advanced#200]
I posted this feature request on airsonic already (https://github.com/airsonic/airsonic/issues/1624) but it may seem interesting here as well:

## Problem description

Podcasts are managed on an application perspective. Different from access to media folders, which can be enabled per user, all users of the application have access to all podcast subscriptions. It would be nice, if different subscriptions could be registered per user, i.e. user A has access to his private podcast subscriptions and user B doesn't see those of user A.

## System information

 * **Airsonic version**: *10.6.0-RELEASE*
 * **Operating system**: *official docker image*
 * **Java version**: *as proved by official docker image*
 * **Proxy server**: *nginx ssl proxy (v. 1.17.9)*
 * **Client**: *firefox (v. 75.0), dsub (v. 5.4.4)*
 * **Language**: *english*

**Comment by randomnicode** [comment]
Fair enough. Haven't dug through the podcast tables, but likely requires:
- an owner (or creator) column
- a public/private column (denotes whether everyone can see it)
- additional column or table for specifying specific users in case you want to only share with targeted users, but not the world (this and above may be combined)

The potential issues that might arise are:
- Two users independently subscribing to the same podcasts (who's the creator? which episodes are kept and which are deleted? how is it managed/orchestrated?)
  - Maybe each subscription is a separate row in the db and the db keeps track of what's "deleted" per channel (i.e. soft-deletes, and keeps on the filesystem)
- Usage of podcast paths as identifiers.
- Need to differentiate between writable podcasts (stuff you own and can manage) and readable podcasts (shared with you), and all the management code (similar to managing playlists)

**Comment by chrstn-hntschl** [comment]
Wouldn't it be simpler to give each user a "podcast library" or folder - just like the existing folders ? So each user can subscribe to podcasts on individual level. If two users subscribe to the same podcast these are redundant. But it also allows users to delete episodes they have listened to. How is file upload handled? If a user has the permission to upload content, who "owns" it? Sorry, if this raises more questions than answers... Maybe these thoughts are helpful as well and in any case many thanks for your investment in this project!

**Comment by randomnicode** [comment]
> Wouldn't it be simpler to give each user a "podcast library" or folder - just like the existing folders ?

That would be another way to do it, but a little "wasteful" in terms of disk use. Though simpler in terms of management (every user has his own podcast "workspace")

Additional considerations with this implementation would have to include a migration strategy. At first migration, does the system create copies of the existing podcasts and duplicate each one of them into every user folder? That might be problematic.

**Comment by chrstn-hntschl** [comment]
Ok fair enough - I haven't considered migration issues t.b.h. because I mainly use the setup with a small amount of users who could be asked to resubscribe to podcasts. I addition, I am currently not using the podcast feature very much, because of the fact that they are not per user... I guess it drills down to the decision whether a possible migration strategy would allow for possible loss in downloaded episodes (I guess the feeds could be more easily migrated) and whether the obvious waste of disk space is acceptable (if you assume that different users share many podcasts ).

**Comment by CHBMB** [comment]
I'd love to see this too.  My wife has no desire to see my Linux Podcasts, and I have no desire to see her gardening one. There is some crossover in our preferences though, so I agree the approach suggested by @randomnicode is sensible.
